### ISSUE TITLE
Training Loss printed looks incorrect.

### ISSUE THREAD

**Issue by yanfeiguan** [chemprop/chemprop#74]
https://github.com/chemprop/chemprop/blob/71e5e86db15df10bbdbb672f8b06dd574f93f6eb/chemprop/train/train.py#L86

The iter_count is updated by the batch_size in each iteration that the printed loss_avg seems like to be much smaller than the actual training loss.

**Comment by swansonk14** [comment]
Hi @yanfeiguan, you're right that `iter_count` is updated by `batch_size` in each iteration, but that's because each time `batch_size` is added, we're adding `loss.item()` to `loss_sum`, where `loss.item()` is the loss for the entire batch. So overall this results in `loss_sum` being the sum of all losses so far and `iter_count` being the number of molecules so far, and so `loss_avg` is the average loss per molecule.

Does that answer your question at all or is there still a difference between the printed loss and the true loss?

**Comment by yanfeiguan** [comment]
Thanks! I think that makes sense, but I am still confused that the loss.item() is the averaged loss for that batch. In that way, the loss/iter_count seems like to be much smaller (real loss/batch_size) to me. In my training process, I also noticed that the training loss is smaller by the fold of batch_size than the validating loss. Please correct me if I was wrong.

**Comment by swansonk14** [comment]
Oh I see now, you're totally right. Since `loss.item()` is already averaged across a batch then `iter_count` should really be the number of batches rather than the number of molecules. Does this commit fix it correctly? https://github.com/chemprop/chemprop/commit/e9a28dea80bc6d255e380dfd10b14549d8c687cc

**Comment by yanfeiguan** [comment]
Yep! It looks correct to me now! Thanks

**Comment by swansonk14** [comment]
Okay great! Thanks again for bringing up the issue!
